examples/ebi-example.js

```diff
@@ -0,0 +1,35 @@
+const { contentsSearch } = require('../lib/ebi');
+
+// NOTE: Assumes you have `GITHUB_PERSONAL_ACCESS_TOKEN` environment variable set
+const { GITHUB_PERSONAL_ACCESS_TOKEN } = process.env;
+
+const repos = [
+	'Financial-Times/ebi',
+	'Financial-Times/tako',
+	'Financial-Times/not-found-error'
+];
+
+async function main() {
+	const allResultsPromises = await contentsSearch({
+		token: GITHUB_PERSONAL_ACCESS_TOKEN,
+		filepath: 'package.json',
+		search: 'ebi'
+	})(repos);
+	const allResults = await Promise.all(
+		allResultsPromises.map(promise => {
+			return promise.catch(e => e);
+		})
+	);
+
+	const searchMatches = allResults.filter(({ type }) => type === 'match');
```

Comment:
This could be more ergonomic, as getting search matches is probably what most people want to do with the `ebi` library ie, don't really care about no matches and errors.

Could add a `contentsSearchMatches`, that just resolves all matches (so only 1 `await` is required, and the user doesn't need a `Promise.all`).

Could do the same for `contentsSearchNoMatches` and `contentsSearchErrors` too.

Example:

```
const searchMatches = await contentsSearchMatches({
	token: GITHUB_PERSONAL_ACCESS_TOKEN,
	filepath: 'package.json',
	search: 'ebi'
})(repos);
console.log(searchMatches);
```

Reply:
Have updated to have a synchronous way (through a function) and asynchronous way to access the results

From `ebi-example.js`:

```
async function packageResults() {
	const { getResults } = await packageSearch({
		token: GITHUB_PERSONAL_ACCESS_TOKEN,
		search: 'ebi'
	})(repos);

	const {
		allResults,
		searchMatches,
		searchNoMatches,
		searchErrors
	} = await getResults();
	console.log('packageSearch: searchMatches', resultsSummary(searchMatches));
	console.log(
		'packageSearch: searchNoMatches',
		resultsSummary(searchNoMatches)
	);
	console.log('packageSearch: searchErrors', resultsSummary(searchErrors));
	console.log('packageSearch: allResults', resultsSummary(allResults));
}

async function packageResultsAsync() {
	const { resultsAsync } = await packageSearch({
		token: GITHUB_PERSONAL_ACCESS_TOKEN,
		search: 'ebi'
	})(repos);

	const allResults = await Promise.all(
		resultsAsync.map(promise => {
			return promise.catch(e => e);
		})
	);

	console.log('packageSearchAsync: allResults', resultsSummary(allResults));
}
```

What do you think @quarterto ?